zipline/pipeline/loaders/blaze/_core.pyx

```diff
@@ -314,6 +312,21 @@ cdef _array_for_column_impl(object dtype,
 
         column_ix = <object> column_ix_ob  # cast to np.int64_t
 
+        with cython.boundscheck(False), cython.wraparound(False):
+            asof_date = asof_dates[n]
+            if asof_date >= most_recent_asof_date_for_ix[asof_ix, column_ix]:
+                # The asof_date is the same or more recent than the
+                # last recorded asof_date at the given index and we should
+                # treat this value as the best known row. We use >=
+                # because a more recent row with the same asof_date
+                # should be treated as an adjustment and the new value
+                # becomes the best-known.
+                most_recent_asof_date_for_ix[asof_ix, column_ix] = asof_date
+            else:
+                # The asof_date is earlier than the asof_date written
+                # at the given index. Ignore this row.
+                continue
```

Comment:
Checking my own comprehension: `continue`-ing here is saying that "if we get records with asof_dates of saturday and sunday, but we only care about the value as of monday, then there's no case where we ever care about the record whose asof is saturday, so we should throw that record in the trash."?

Reply:
Correct, we didn't know about the Saturday record on Friday, and by the time Monday comes we have more up to date information. There is no row in the output where that record would be the most recently known data for the time.